pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -311,6 +313,7 @@ func (asw *actualStateOfWorld) AddVolumeNode(
 			nodeName)
 	}
 
+	// TODO: Is this a full attach?  Maybe only if we have a device path?
```

Comment:
The purpose of this function is to keep a record VolumeToReportAsAttached list in actual state of reconciler on master indicating this volume is attached to the node (during detaching, this record will be deleted from the list). The list will be used by master's node status updater to update the Node.Status.VolumesAttached on API server so that kubelet on the node can retrieve this list and get to know which volume is already attached and what is the device path. In another word, the information is used to pass from master to node.